refnx/reduce/platypusnexus.py

```diff
@@ -671,111 +705,129 @@ def __init__(
-        mbf = ManualBeamFinder()
-
-        spinset = SpinSet(
-            "PLP0051296.nx.hdf",
-            "PLP0051294.nx.hdf",
-            up_down="PLP0051295.nx.hdf",
-            down_up="PLP0051297.nx.hdf",
-            data_folder=data_dir
-        )
-        spinset.du_opts.update({"manual_beam_find" : mbf, peak_pos : -1})
-        spinset.process_beams()
+        If reduction_options is None, the reduction options for each spin
+        channel are specified by the dictionary of spin channel reduction options SpinSet.sc_opts which are initialised to the
+        standard options when constructing the object. If `reduction_options`
+        is not None, then SpinSet.process() will use these options for all
+        spin channels.
 
         Parameters
         ----------
         reduction_options : dict
-            A single dict of options used to process all spectra. If
-            this is None, then process_beams will use individual dicts
-            for each spin channel
+            A single dict of options used to process all spectra.
         """
-        # TODO consider removing this method, not clear how it's going to be
-        # used.
-        if reduction_options:
+
+        if reduction_options is not None:
             print(
                 "Applying the supplied reduction_options to all spin channels"
             )
-            self.dd_opts = reduction_options.copy()
-            self.du_opts = reduction_options.copy()
-            self.ud_opts = reduction_options.copy()
-            self.uu_opts = reduction_options.copy()
-
-        for opts, beam in zip(
-            [self.dd_opts, self.du_opts, self.ud_opts, self.uu_opts],
-            [self.dd, self.du, self.ud, self.uu],
-        ):
-            if beam is None:
+            for sc in self.sc_opts:
+                sc = reduction_options.copy()
+
+        # Check important reduction options are the same across all
+        # spin channels to ensure the same wavelength axis
+
+        for key in [
+            "lo_wavelength",
+            "hi_wavelength",
+            "rebin_percent",
+            "wavelength_bins",
+        ]:
+            for option1 in ["dd", "du", "ud", "uu"]:
+                for option2 in ["dd", "du", "ud", "uu"]:
+                    if self.sc_opts[option1] is None:
+                        continue
+                    elif self.sc_opts[option2] is None:
+                        continue
+                    elif self.sc_opts[option1] != self.sc_opts[option2]:
+                        raise ValueError(
+                            "Reduction options `lo_wavelength`, "
+                            "`hi_wavelength`, `rebin_percent`, and "
+                            "`wavelength_bins` must be identical across "
+                            "spin channels to preserve a common "
+                            "wavelength axis."
+                        )
+
+        for sc in ["dd", "du", "ud", "uu"]:
```

Comment:
I think we still need to check that the spin channel isn't None (wasn't measured) since the `down_up` and `up_down` keys are present in `self.spin_channels` (would you prefer to have a `measured_channels` parameter to minimise the number of these checks?).

 I have currently done the following:

```
for sc, channel in self.channels:
    if channel is None:
        continue
    else:
        channel.process(**self.sc_opts[sc])
```